pkg/store/multitsdb.go

```diff
@@ -122,9 +122,14 @@ func newTenantSeriesSetServer(
 
 func (s *tenantSeriesSetServer) Context() context.Context { return s.ctx }
 
+type clientSpanTagKey struct {
+	tenant string
+}
+
 func (s *tenantSeriesSetServer) Series(store storepb.StoreServer, r *storepb.SeriesRequest) {
 	var err error
-	tracing.DoInSpan(s.ctx, "multitsdb_tenant_series", func(_ context.Context) {
+	tracing.DoInSpan(s.ctx, "multitsdb_tenant_series", func(ctx context.Context) {
+		ctx = context.WithValue(ctx, clientSpanTagKey{tenant: s.tenant}, s.tenant)
```

Comment:
It might work but you are making it too complex. Can you just add a tag like https://github.com/thanos-io/thanos/blob/main/pkg/compact/compact.go#L773?

Reply:
Thanks, I had initially looked at the grpc-opentracing codebase and used the method of instrumentation referring to that.